From af6e9b896aa4959c721d49246a56e63ceb1d695b Mon Sep 17 00:00:00 2001
From: Victor Morales <victor.morales@intel.com>
Date: Thu, 8 Oct 2015 10:24:32 -0500
Subject: [PATCH] Use oslo.config to search for config files

Make wsgi.Loader class use oslo_config CONF.find_file function rather than
utils.find_config

Remove utils.find_config function and its test cases to avoid code duplication

Modify unit tests to look for config files in etc/cinder since this is not a
default location for oslo_config to look for

Signed-off-by: Victor Morales <victor.morales@intel.com>
---
 cinder/wsgi/common.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cinder/wsgi/common.py b/cinder/wsgi/common.py
index 2fd4e6a..baae151 100644
--- a/cinder/wsgi/common.py
+++ b/cinder/wsgi/common.py
@@ -267,7 +267,10 @@ class Loader(object):
 
         """
         config_path = config_path or CONF.api_paste_config
-        self.config_path = utils.find_config(config_path)
+        self.config_path = CONF.find_file(config_path)
+
+        if not self.config_path:
+            raise exception.ConfigNotFound(path=config_path)
 
     def load_app(self, name):
         """Return the paste URLMap wrapped WSGI application.
-- 
2.1.0

